lib/util.py

```diff
@@ -542,55 +542,62 @@ def format_time(timestamp):
 # Takes a timestamp and returns a string with the approximation of the age
 def age(from_date, since_date = None, target_tz=None, include_seconds=False):
     if from_date is None:
-        return "Unknown"
+        return _("Unknown")
 
     from_date = datetime.fromtimestamp(from_date)
     if since_date is None:
         since_date = datetime.now(target_tz)
+    else:
+        if isinstance(since_date, int) or isinstance(since_date, float):
+            since_date = datetime.fromtimestamp(since_date)
 
     td = time_difference(from_date - since_date, include_seconds)
-    return td + " ago" if from_date < since_date else "in " + td
+
+    if from_date < since_date:
+        return _("{time} ago").format(time=td)
+    else:
+        return _("in {time}").format(time=td)
 
 
 def time_difference(distance_in_time, include_seconds):
-    #distance_in_time = since_date - from_date
+    #distance_in_time = from_date - since_date
     distance_in_seconds = int(round(abs(distance_in_time.days * 86400 + distance_in_time.seconds)))
-    distance_in_minutes = int(round(distance_in_seconds/60))
+    distance_in_minutes = int(round(distance_in_seconds / 60))
 
-    if distance_in_minutes <= 1:
+    if distance_in_seconds < 60:
         if include_seconds:
             for remainder in [5, 10, 20]:
                 if distance_in_seconds < remainder:
-                    return "less than %s seconds" % remainder
+                    return _("less than {seconds} seconds").format(seconds=remainder)
             if distance_in_seconds < 40:
-                return "half a minute"
-            elif distance_in_seconds < 60:
-                return "less than a minute"
+                return _("half a minute")
             else:
-                return "1 minute"
+                return _("less than a minute")
         else:
-            if distance_in_minutes == 0:
-                return "less than a minute"
-            else:
-                return "1 minute"
+            return _("less than a minute")
     elif distance_in_minutes < 45:
-        return "%s minutes" % distance_in_minutes
+        fmt = ngettext("{minutes} minute", "{minutes} minutes", distance_in_minutes)
```

Comment:
It kind of bugs me that from 61 seconds to 89 seconds here it will say `1 minute` which implies some degree of precision.. but in other places we say `about X_TIME`.  Hmmm.... would it be a good idea to say `about X minute(s)` here too?  (And in all other places that rounding is basically used?)

Reply:
Yea, I changed to to "about a minute".